### PR TITLE
Towards v2

### PR DESCRIPTION
**Pull request by tbaudier**

### CODE REVIEW

File: spyrit/learning/imagenet_dataset.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+from PIL import Image
+import os
```
Comment (tbaudier) [inline spyrit/learning/imagenet_dataset.py:2]: File already in towards_v2_fadoua

File: spyrit/learning/model_Had_DCAN.py [hunk 2]
```diff
@@ -685,4 +685,40 @@ def forward(self,x,y, net):
         mse = self.loss(x,y);
         return mse
 
+class Variance_Loss(nn.Module):
+    def __init__(self, loss, mean_net):
```
Comment (tbaudier) [inline spyrit/learning/model_Had_DCAN.py:689]: present in towards_v2_fadoua

File: spyrit/reconstruction/kalman.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import torch
+import numpy as np
+import torchvision
+from torchvision import datasets, models, transforms
```
Comment (tbaudier) [inline spyrit/reconstruction/kalman.py:4]: present in towards_v2_fadoua

File: spyrit/reconstruction/recon_functions.py [hunk 4]
```diff
@@ -0,0 +1,815 @@
+import math
+import torch
+import torch.nn as nn
```
Comment (tbaudier) [inline spyrit/reconstruction/recon_functions.py:3]: present in towards_v2_fadoua

File: spyrit/restructured/Updated_Had_Dcan.py [hunk 5]
```diff
@@ -411,20 +411,29 @@ def sigma(self, x):
...
         sigdark: standard deviation or dark current in counts
         nbin: number of raw bin in each spectral channel (if input x results 
         from the sommation/binning of the raw data)
         """
+        self.gain = gain
+        self.mudark = mudark
+        self.sigdark = sigdark
+        self.nbin = nbin
+    
+    def sigma_expe(self, x):
```
Comment (tbaudier) [inline spyrit/restructured/Updated_Had_Dcan.py:429]: present in towards_v2_fadoua in core/AI.py

**Comment by tbaudier** [comment]
For me, Towards_v2 branch can be deleted because all modifications are in Towards_v2_fadoua

**Comment by nducros** [comment]
OK let's go!

**Pull request closed without merging**